synapseclient/client.py

```diff
@@ -6031,6 +6036,49 @@ def get_membership_status(self, userid, team):
 
         Returns:
             dict of TeamMembershipStatus
+
+        Example: Using this function (DEPRECATED)
+        Getting a user's membership status for a team
+
+            status = syn.get_membership_status(user="12345", team="67890")
+            print(status)
+
+
+        Example: Migration to new method
+            &nbsp;
+
+            ```python
+            from synapseclient import Synapse
+            from synapseclient.models import Team
+
+            # Login to Synapse
+            syn = Synapse()
+            syn.login()
+
+            # Use synchronous version (recommended for most use cases)
+            team = Team.from_id(id="67890")
+            status = team.get_user_membership_status(
+                user_id="12345",
+            )
+            print(status)
+
+            # Alternative: Use async version for advanced use cases
+            import asyncio
+
+            async def get_membership_status():
+                # Get the team object
+                team = await Team.from_id_async(id="67890")
+
+                # Get user's membership status
+                status = await team.get_user_membership_status_async(
+                    user_id="12345"
+                )
+                return status
+
+            # Run the async function
+            status = asyncio.run(get_membership_status())
+            print(status)
```

Comment:
tbh I probably wouldn't even include the async version in the migration note. It is more complete to include it, but may confuse someone coming in and looking at this.